UrlManager.php

```diff
@@ -370,6 +382,14 @@ protected function processLocaleUrl($normalized)
             if ($this->enableLanguagePersistence) {
                 $language = $this->loadPersistedLanguage();
             }
+            if ($language===null && isset($_SERVER[$this->geoipServerFieldName]) && !empty($this->geoipLanguageCountries)) {
```

Comment:
I would say, you could remove `!empty($this->geoipLanguageCountries)` here because if the array is empty, `foreach` will be skipped anyway.